### PR TITLE
Many Sphinx fixes (including a fix of the credits chapter)

### LINKED ISSUE #7209: Update credits following Sphinx migration
`doc/sphinx/credits.rst` and `CREDITS` should contain @cpitclaudel's name.
`doc/sphinx/credits.rst` should probably also list all the people who helped with the porting as a way of thanking them. They don't need to be in `CREDITS` because helping with porting from one format to another doesn't make you a copyright holder.
This is a blocker for 8.8.0.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** documentation fix

Fixes / closes #7209.

~~@cpitclaudel Can you review 6c5de0ee9a823c0536ff2c1e8cdf3f3282f515b0?~~ Done

### CODE REVIEW

**Comment by mattam82** [comment]
The credits part looks fine to me. Maybe rephrase "The infrastructure..."? It sounds a bit unnatural to me.

**Review by maximedenes — APPROVED** [review]
Looks very good, thanks!
